core/src/main/kotlin/org/evomaster/core/search/StructuralElement.kt

```diff
@@ -142,6 +148,7 @@ abstract class StructuralElement (
 
     fun addChildToGroup(position: Int, child: StructuralElement, groupId: String){
         verifyChildrenToInsert(child)
+        preChildrenSetup(listOf(child))
```

Comment:
shouldn't this be done after checking validity of inputs?

Reply:
This was changed by @ZhangMan1126

Reply:
I pushed a change, in which I moved it after the input validation (as well for the other cases)